src/iminuit/qtwidget.py

```diff
@@ -0,0 +1,413 @@
+            elif self.val > self.vmax:
+                self.vmax = self.val
+                self.tmax.blockSignals(True)
+                self.tmax.setValue(self.vmax)
+                self.tmax.blockSignals(False)
+
+            self.slider.blockSignals(True)
+            self.slider.setValue(self._float_to_int(self.val))
+            self.slider.blockSignals(False)
+            self.value_label.setText(f"{self.val:.3g}")
+
+    class MainWindow(QtWidgets.QMainWindow):
+        def __init__(self):
+            super().__init__()
+            self.resize(1200, 600)
+            font = QtGui.QFont()
+            font.setPointSize(12)
+            self.setFont(font)
+            centralwidget = QtWidgets.QWidget(parent=self)
+            self.setCentralWidget(centralwidget)
+            central_layout = QtWidgets.QVBoxLayout(centralwidget)
+            tab = QtWidgets.QTabWidget(parent=centralwidget)
+            interactive_tab = QtWidgets.QWidget()
+            tab.addTab(interactive_tab, "Interactive")
+            results_tab = QtWidgets.QWidget()
+            tab.addTab(results_tab, "Results")
+            central_layout.addWidget(tab)
+
+            interactive_layout = QtWidgets.QGridLayout(interactive_tab)
+
+            plot_group = QtWidgets.QGroupBox("", parent=interactive_tab)
+            size_policy = QtWidgets.QSizePolicy(
+                QtWidgets.QSizePolicy.Policy.MinimumExpanding,
+                QtWidgets.QSizePolicy.Policy.MinimumExpanding,
+            )
+            plot_group.setSizePolicy(size_policy)
+            plot_layout = QtWidgets.QVBoxLayout(plot_group)
+            # Use pyplot here to allow users to use pyplot in the plot
+            # function (not recommended / unstable)
+            self.fig, ax = plt.subplots()
+            self.canvas = FigureCanvasQTAgg(self.fig)
+            plot_layout.addWidget(self.canvas)
+            plot_layout.addStretch()
+            interactive_layout.addWidget(plot_group, 0, 0, 2, 1)
+            try:
+                plot(minuit.values, fig=self.fig)
+                kwargs["fig"] = self.fig
+            except Exception:
+                pass
+            try:
+                plot(minuit.values, ax=ax)
+                kwargs["ax"] = ax
+            except Exception:
+                pass
+            self.fig_width = self.fig.get_figwidth()
+
+            button_group = QtWidgets.QGroupBox("", parent=interactive_tab)
+            size_policy = QtWidgets.QSizePolicy(
+                QtWidgets.QSizePolicy.Policy.Expanding,
+                QtWidgets.QSizePolicy.Policy.Fixed,
+            )
+            button_group.setSizePolicy(size_policy)
+            button_layout = QtWidgets.QHBoxLayout(button_group)
+            self.fit_button = QtWidgets.QPushButton("Fit", parent=button_group)
+            self.fit_button.setStyleSheet("background-color: #2196F3; color: white")
+            self.fit_button.clicked.connect(partial(self.do_fit, plot=True))
```

Comment:
A lambda is as terse and faster to execute for the Python interpreter.

Reply:
It is better to use the basic features of the language instead of library functions, if those do not add anything essential.

Reply:
agreed [e20f357](https://github.com/scikit-hep/iminuit/pull/1056/commits/e20f357b5734260ac69a74a0e50287850bd44d19)